packages/venia-concept/src/components/Main/main.js

```diff
@@ -33,9 +33,9 @@ class Main extends Component {
 
         return (
             <main className={classes.root}>
-                <Header />
-                <div className={classes.page}>{children}</div>
-                <Footer />
+                <Header style={{ flex: 1 }} />
```

Comment:
This is not ideal, but it was necessary to fix a bad stretch problem with the parent container that was blocking my media query tests.

I need to inject styles into the `Header` from the parent component. If `classify` allowed us to compose classnames by joining them with spaces, then I could inject extra styles into `Header` just by knowing the classnames, e.g.

main.css:
```css
.mainheader {
  flex: 1
}
```

```jsx
<Header classes={{ header: mainheader }} />
```

Reply:
https://github.com/magento-research/pwa-studio/blob/release/2.0/packages/venia-concept/src/components/Header/header.js#L46

I think I've done something similar in custom theme. From this file, we can see that either `classes.open` or `classes.closed` will be applied to the Header HTML element. So you can override these 2 classes from parent by doing the following:

- `<Header classes={{ open: classes.mainheaderOpen, close:  classes.mainheaderClose}} />`
- In the main.css file where mainheaderOpen and mainheaderClose are located, compose the class from Header component (`open` and `close` respectively)

Let me know if this works. I will check tomorrow directly on your branch just to be sure and get back to you.

Reply:
To follow up on my suggestion, I have checked this locally and it works. Let me know if this solution is alright:

This is an example of overriding header class only. Same technique can be applied for others.

main.css
```css
.headerOverride {
    composes: root from 'src/components/Header/header.css';
    flex: 1;
}

```
main.js
```jsx
<Header
     classes={{
          open: this.props.classes.headerOverride,
          closed: this.props.classes.headerOverride
     }}
/>
```

Note: both close and open classes inherit from root class in header.css, so I think that they can be safely removed.

Result:

![screen shot 2019-01-15 at 2 50 59 pm](https://user-images.githubusercontent.com/11479290/51184587-00adc680-18d5-11e9-819b-fb865f7404f0.png)